router/usecase/pools/spot_price_quote_calculator_test.go

```diff
@@ -0,0 +1,230 @@
+package pools_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/osmosis-labs/osmosis/osmomath"
+	"github.com/osmosis-labs/sqs/domain"
+	"github.com/osmosis-labs/sqs/router/usecase/pools"
+	"github.com/osmosis-labs/sqs/router/usecase/routertesting"
+)
+
+// This test case validates that the spot price quote calculator computes
+// the spot price via quote as intended and returns errors in edge cases
+// as given by the spec.
+//
+// To setup the environment, it configures 2 mocks:
+// 1. Mock quote estimator callback
+// 2. Scaling factor computer callback.
+//
+// Then, by using the mocked outputs, it validates the behavior to be as intended.
+// For details about mock configuration, read their respective setup functions.
+func (s *RoutablePoolTestSuite) TestSpotPriceQuoteCalculator_Calculate() {
```

Comment:
The mock setup below looks a bit bulky. Does it make sense to move the mock things to domain/mocks ?

Reply:
I would agree if these mocks were meant to be reused across the system.

However, they are intended to be specific for this test.

As a result, to avoid polluting the test-global scope, I propose keeping them here despite the bulkiness